url882f91ec97_WenboCao_Microsoft_Drone_tgz-pJ8-org_ejml_alg_dense_decomposition_qr_StabilityQRDecompositionJ8/src/org/ejml/alg/dense/misc/GenerateInverseFromMinor.java

```diff
@@ -0,0 +1,240 @@
+/*
+ * Copyright (c) 2009-2013, Peter Abeles. All Rights Reserved.
+ *
+ * This file is part of Efficient Java Matrix Library (EJML).
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.ejml.alg.dense.misc;
+
+import org.ejml.alg.generic.CodeGeneratorMisc;
+
+import java.io.FileNotFoundException;
+import java.io.PrintStream;
+
+
+/**
+ * Generates unrolled matrix from minor analytical functions.  these can run much faster than LU but will only
+ * work for small matrices.
+ *
+ * When computing the determinants for each minor there are some repeat calculations going on.  I manually
+ * removed those by storing them in a local variable and only computing it once.  Despite reducing the FLOP count
+ * it didn't seem to noticeably improve performance in a runtime benchmark..
+ *
+ * @author Peter Abeles
+ */
+public class GenerateInverseFromMinor {
+
+    String className = "UnrolledInverseFromMinor";
+    PrintStream stream;
+    int N;
+
+    public GenerateInverseFromMinor( boolean createStream ) throws FileNotFoundException {
+        if( createStream )
+            stream = new PrintStream(className +".java");
```

Comment:
Recommendation generated by Amazon CodeGuru Reviewer. Leave feedback on this recommendation by replying to the comment or by reacting to the comment using emoji.

**Problem**
This line of code assigns a resource to a class field and there is no method that closes or returns this resource. This can result in a resource leak that slows down or crashes your system.

**Fix**
Create a method in this class that closes the following resource: *stream*. After you use this resource, use this method to prevent a resource leak.

**More info**
[View resource management guidelines at oracle.com](https://www.oracle.com/technetwork/articles/java/trywithresources-401775.html) (external link).